TcBaseTest/TcBaseTest/TcBaseTest/State Pattern Test/FB_StatePattern_Test.TcPOU

```diff
@@ -0,0 +1,269 @@
+VAR_INPUT
+END_VAR
+]]></Declaration>
+      <Implementation>
+        <ST><![CDATA[TcUnit.TEST_ORDERED('TestClearAllStatesExpectSuccess');
+
+THIS^.fbStateMachine1.ClearStates();
+
+THIS^.AssertEquals_UDINT(Expected := 0,
+                         Actual := THIS^.fbStateMachine1.nStateCount,
+                         Message := 'State machine did not clear all states correctly');
+
+THIS^.AssertFalse(Condition := THIS^.fbStateMachine1.Contains(THIS^.fbState11),
+                  Message := 'Cleared state should not be in state machine');
+THIS^.AssertFalse(Condition := THIS^.fbStateMachine1.Contains(THIS^.fbState12),
+                  Message := 'Cleared state should not be in state machine');
+THIS^.AssertFalse(Condition := THIS^.fbStateMachine1.Contains(THIS^.fbState13),
+                  Message := 'Cleared state should not be in state machine');
+THIS^.AssertFalse(Condition := THIS^.fbStateMachine1.Contains(THIS^.fbState14),
+                  Message := 'Cleared state should not be in state machine');
+
+TcUnit.TEST_FINISHED();
+]]></ST>
+      </Implementation>
+    </Method>
+    <Method Name="TestSeasonStateMachineAllSeasons" Id="{78f75ccf-d3ea-4cf4-a9dd-45ff9652d3b6}">
+      <Declaration><![CDATA[METHOD PRIVATE TestSeasonStateMachineAllSeasons
+VAR_INPUT
+END_VAR
+]]></Declaration>
+      <Implementation>
+        <ST><![CDATA[TcUnit.TEST('TestSeasonStateMachineAllSeasons');
+
+THIS^.fbYear.Run();
+
+// Summer
+THIS^.AssertEquals_BYTE(Expected := 1,
+                        Actual := THIS^.fbYear.CurrentState().nTyresType,
+                        Message := 'Tyres initialization is incorrect for Summer');
+THIS^.AssertTrue(Condition := THIS^.fbYear.CurrentState().Equals(THIS^.fbYear.StateSummer),
+                 Message := 'Expected Summer to be the starting season');
+THIS^.AssertTrue(Condition := THIS^.fbYear.CurrentState().bSunShining,
+                 Message := 'Expected the sun to shine during Summer');
+THIS^.AssertEquals_BYTE(Expected := 1,
+                        Actual := THIS^.fbYear.CurrentState().nLeavesState,
+                        Message := 'Expected the color of the leaves to be green');
+THIS^.fbYear.ChangeToWinterTyres();
+THIS^.AssertEquals_BYTE(Expected := 1,
+                        Actual := THIS^.fbYear.CurrentState().nTyresType,
+                        Message := 'Must not be able to put winter tyres on during Summer');
+
+// Autumn
+THIS^.fbYear.NextSeason();
+THIS^.AssertTrue(Condition := THIS^.fbYear.CurrentState().Equals(THIS^.fbYear.StateAutumn),
+                 Message := 'Expected Autumn to be after Summer');
+THIS^.AssertTrue(Condition := THIS^.fbYear.CurrentState().bWindBlowing,
+                 Message := 'Expected the wind to blow during Autumn');
+THIS^.AssertEquals_BYTE(Expected := 2,
+                        Actual := THIS^.fbYear.CurrentState().nLeavesState,
+                        Message := 'Expected the color of the leaves to be (changing to) yelow/orange/red');

```

Comment:
small typo on yellow :)

Reply:
To be fixed in later commits to avoid merge conflicts, since there has been more development since this PR